### PR TITLE
Integrate Alerting Dashboards with Notifications Plugin

### PR DESCRIPTION
**Pull request by qreshi**
### Description
Integrates Alerting Dashboards with the Notifications plugin:
* Adds a banner explaining the deprecation of Destinations and a link to the Notifications plugin dashboard
* Disabled buttons for removed actions related to Destinations
* Supports selecting Notification Channels when creating an Action and denotes the difference between `Channels` and `Destinations` to support pre-migration scenarios

#### Follow-ups that need to be done in a subsequent PR:
* Add a reference link in the `"Learn more"` message for Notifications when the Notifications documentation is available
* Remove the disabled `Add destination` button
* Remove the disable `Add sender` and `Add group` buttons
* Remove the `delete` button for Destinations
* Change the `edit` button for Destinations to `view` and make the fields read-only on the Destination page
* Remove any references to `Update` when viewing read-only Destinations and change `Cancel` to be `Close`
* Confirm if the Channel selection when creating an Action is disabled when Notifications is disabled and add a message there when it is
 
### Issues Resolved
Does not completely resolve but related to https://github.com/opensearch-project/alerting/issues/105
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/services/utils/helper.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  };
+};
+
+export const configListToChannels = (configs: any[]): ChannelItemType[] => {
+  return configs?.map((config) => configToChannel(config)) || [];
+};
+
+// export const configToSender = (config: any): SenderItemType => {
+//   return {
+//     name: config.config.name,
```
Comment (lezzago) [inline public/services/utils/helper.ts:28]: Can we remove these comments?

File: public/services/utils/helper.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChannelItemType,
+  // NotificationItem,
+  // RecipientGroupItemType,
+  // SenderItemType,
```
Comment (lezzago) [inline public/services/utils/helper.ts:10]: Can we remove these comments?

**Review by lezzago — APPROVED** [review]
LGTM, just minor comments

File: public/services/utils/helper.ts (hunk 1 above)
Comment (qreshi) [reply public/services/utils/helper.ts:28]: Will do a clean-up passthrough in a follow-up

**Review by praveensameneni — APPROVED** [review]

**Review by lezzago — APPROVED** [review]

**Review by praveensameneni — APPROVED** [review]

**Pull request merged**